### PR TITLE
Feature/documentation up

### PR DESCRIPTION
**Pull request by Jhoneagle**
Javadoc update and generalized naming a bit.

### CODE REVIEW

File: src/main/java/fi/helsinki/ubipositioning/datamodels/package-info.java [hunk 1]
```diff
@@ -1,4 +1,6 @@
 /**
- * Package to contain all data models needed for trilateration.
+ * Data models used for representing BLE listeners, BLE devices
+ * and spesific locations with extra information related to
```
Comment (je-l) [inline src/main/java/fi/helsinki/ubipositioning/datamodels/package-info.java]: spesific -> specific

File: src/main/java/fi/helsinki/ubipositioning/mqtt/MqttService.java [hunk 2]
```diff
@@ -10,49 +10,64 @@
...
  */
-public class MqttService implements IMqttService{
-    public static final int MAX_OBSERVATIONS = 10000;
+public class MqttService implements IMqttService {
+    /**
+     * Default buffer for storing data that comes from subscribed topic.
+     */
+    private static final int MAX_MESSAGES_TO_STORE = 10000;
+    /**
+     * Default lifetime for beacons.
```
Comment (je-l) [inline src/main/java/fi/helsinki/ubipositioning/mqtt/MqttService.java]: I think we could clarify that this is seconds.

File: src/main/java/fi/helsinki/ubipositioning/mqtt/MqttService.java [hunk 3]
```diff
@@ -10,49 +10,64 @@
...
-     * @param mqttUrl Mqtt server URL.
-     * @param subTopic Topic to subscribe.
-     * @param pubTopic Topic to publish.
-     * @param beacons initial Beacon information.
+     * Creates instance of the class.
+     *
+     * @param mqttUrl Mqtt servers URL.
+     * @param subTopic Topic to subscribe to.
+     * @param pubTopic Topic to publish messages to.
+     * @param beacons Initial Beacon information.
```
Comment (je-l) [inline src/main/java/fi/helsinki/ubipositioning/mqtt/MqttService.java]: Does it matter if we give `mqtt://example.com` or `example.com` in the mqttUrl argument? I think we could give example of the mqtt url here, if both formats are not accepted.

Also i think we should clarify that subTopic is the source of RSSI data, and pubTopic is the destination of calculated location data.

File: src/main/java/fi/helsinki/ubipositioning/trilateration/LocationService.java [hunk 4]
```diff
@@ -9,49 +8,58 @@
 import fi.helsinki.ubipositioning.datamodels.*;
 import fi.helsinki.ubipositioning.utils.IObserverService;
 
+/**
+ * Base logic for trilateration calculation that find solution if possible in n-dimensional space.
```
Comment (je-l) [inline src/main/java/fi/helsinki/ubipositioning/trilateration/LocationService.java]: find -> finds

File: src/main/java/fi/helsinki/ubipositioning/utils/ObservationGenerator.java [hunk 5]
```diff
@@ -9,33 +9,54 @@
...
+     */
     public ObservationGenerator(int maxBeacons, int newOnesPerCall, List<String> observerKeys) {
         this.maxBeacons = maxBeacons;
         this.newOnesPerCall = newOnesPerCall;
         this.observerKeys = observerKeys;
+        beacons = new HashMap<>();
     }
 
+    /**
+     * Initializes the mock generator but gives default values for amount of BLE devices can be.
```
Comment (je-l) [inline src/main/java/fi/helsinki/ubipositioning/utils/ObservationGenerator.java]: This sentence seems odd

**Review by je-l — CHANGES_REQUESTED** [review]
Overall looks very good! I left few comments regarding grammar, please merge this after you've resolved those how you see best.

File: src/main/java/fi/helsinki/ubipositioning/mqtt/MqttService.java (hunk 2 above)
Comment (Jhoneagle) [reply src/main/java/fi/helsinki/ubipositioning/mqtt/MqttService.java]: MAX_MESSAGES_TO_STORE or BEACON_LIFETIME?

Comment (je-l) [reply src/main/java/fi/helsinki/ubipositioning/mqtt/MqttService.java]: BEACON_LIFETIME

**Pull request merged**